File: KeyboardTextFieldDemo/IQKeyBoardManager/IQNSArray+Sort.m
//
//  NSArray+Sort.m
// https://github.com/hackiftekhar/IQKeyboardManager
// Copyright (c) 2013-14 Iftekhar Qurashi.
//
// Permission is hereby granted, free of charge, to any person obtaining a copy
// of this software and associated documentation files (the "Software"), to deal
// in the Software without restriction, including without limitation the rights
// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
// copies of the Software, and to permit persons to whom the Software is
// furnished to do so, subject to the following conditions:
//
// The above copyright notice and this permission notice shall be included in
// all copies or substantial portions of the Software.
//
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
// THE SOFTWARE.

#import "IQNSArray+Sort.h"

#import <UIKit/UIView.h>

#import "IQKeyboardManagerConstantsInternal.h"

IQ_LoadCategory(IQNSArraySort)


@implementation NSArray (IQ_NSArray_Sort)

- (NSArray*)sortedArrayByTag
{
    return [self sortedArrayUsingComparator:^NSComparisonResult(UIView *obj1, UIView *obj2) {
        
        if ([obj1 respondsToSelector:@selector(tag)] && [obj2 respondsToSelector:@selector(tag)])
        {
            if ([obj1 tag] < [obj2 tag])	return NSOrderedAscending;
            
            else if ([obj1 tag] > [obj2 tag])	return NSOrderedDescending;
            
            else	return NSOrderedSame;
        }
        else
            return NSOrderedSame;
    }];
}

@end

File: IQKeybordManagerSwift/IQNSArray+Sort.swift
//
//  IQNSArray+Sort.swift
//  IQKeyboard
//
//  Created by Iftekhar on 21/09/14.
//  Copyright (c) 2014 Iftekhar. All rights reserved.
//

import Foundation

extension NSArray {
    
    
    func sortedArrayByTag() -> NSArray {
        
        return self.sortedArrayUsingComparator({ (let view1: AnyObject?, let view2: AnyObject?) -> NSComparisonResult in

            if view1?.tag < view2?.tag {
                return NSComparisonResult.OrderedAscending
            }
            else if view1?.tag > view2?.tag {
                    return NSComparisonResult.OrderedDescending
            }
            else {
                return NSComparisonResult.OrderedSame
            }
        })
    }
    
}